variant_to_realignment.nf

```diff
@@ -17,10 +17,18 @@ process convertVCFToBed {
     output:
         path "variants.bed", emit: variants_bed
 
-    """
-    # TODO: Change vcf2bed so it does not split the alternates into two lines
-    vcf2bed < source.vcf > variants.bed
-    """
+    '''
+    # Convert the vcf file to bed format:
+    #  - Remove all headers
+    #  - Switch to 0 based coordinates system
+    #  - Add the reference allele so it can be used in flankingRegionBed to adjust the position of the right flank
+    #  - add all VCF fields separated by pipe (|)
```

Comment:
On a second read, this is the only thing which still bothers me. A pipe symbol is commonly used in VCF to cram structured data into a string. For example, ClinVar:

```
1	925952	1019397	G	A	.	.	ALLELEID=1003021;CLNDISDB=MedGen:CN517202;CLNDN=not_provided;CLNHGVS=NC_000001.11:g.925952G>A;CLNREVSTAT=criteria_provided,_single_submitter;CLNSIG=Uncertain_significance;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;GENEINFO=SAMD11:148398;MC=SO:0001583|missense_variant;ORIGIN=1
```

I suspect that the `MC=SO:0001583|missense_variant` part will break this approach.

I don't see an easy solution to this. Using multiple pipes, e.g. `|||`, wouldn't be reliable either, because sometimes these structured values have quite a large number of fields, and some of them could be empty, and we can get something like `A|TRUE|0|||2||3.0`.

For now, as a quick fix, this could be perhaps addressed by using a multi-character separator which is unlikely appear in a real life VCF; so something mostly random.

And ultimately, I think it would be a good idea to have a deep refactor of this whole section of the code, so that the VCF records don't have be stored in any other format (BED/BAM), and some different approach to pair the records is used instead. But this is clearly a lot of work and out of scope of this PR, and even perhaps the foreseeable future.

So, I leave the decision on how to handle this thing for you; other than that the changes look great :)

Reply:
That's a good point. It does not cost much more to use a two character delimiter for now so I've implemented that.
I already implemented a [lookup system](https://github.com/EBIvariation/variant-remapping/blob/2abc2d0972a917fb603c5c01ef1cd74ac2850cce/variant_to_realignment.nf#L26) based on `seek` in this PR before, but it turned out to be quite slow.
Longer term I agree that a major refactor (or rewrite) is needed to get around this issue.